File: ACWing/868.cpp
/**
 *    author: subobo
 *    created: 24.07.2022 16:07:20
**/
#include <bits/stdc++.h>

using namespace std;

#ifdef LOCAL
#include "algo/debug.h"
#else
#define debug(...) 42
#endif

vector<int> least = {0, 1};
vector<int> primes;

void RunLinearSieve(int n) {
  n = max(n, 1);
  least.assign(n + 1, 0);
  primes.clear();
  for (int i = 2; i <= n; i++) {
    if (least[i] == 0) {
      least[i] = i;
      primes.push_back(i);
    }
    for (int x : primes) {
      if (x > least[i] || i * x > n) {
        break;
      }
      least[i * x] = x;
    }
  }
}

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int n;
  cin >> n;
  RunLinearSieve(n);
  cout << primes.size() << '\n';
  return 0;
}

File: ACWing/868.py
# author: subobo
# created: 24.07.2022 16:44:32

# def sieve(n):
#   used = [False] * (n + 1)
#   primes = []
#   for c in range(2, n + 1):
#     if not used[c]:
#       primes.append(c)
#       d = c
#       while d <= n:
#         used[d] = True
#         d += c
#   return len(primes)
  
# print(sieve(int(input())))

def sieve(n):
  least = [0] * (n + 1)
  primes = []
  for c in range(2, n + 1):
    if least[c] == 0:
      least[c] = c
      primes.append(c)
      
    for x in primes:
      if x > least[c] or c * x > n:
        break
      least[c * x] = c
  return len(primes)
      
print(sieve(int(input())))